src/python/build_management/revisions.py

```diff
@@ -316,13 +316,18 @@ def get_components_list(component_revisions_dict, job_type):
     # No project name found in job environment, return list as-is.
     return components
 
-  project_src = '/src/%s' % project_name
-  for component in components:
+  project_src = '/src/' + project_name
+  for component in components.copy():
     if component == project_src:
       components.remove(component)
       components.insert(0, component)
       break
 
+    if project_name in os.path.basename(component):
```

Comment:
Should this be startswith to be avoid unexpected matches ?

Reply:
Want to catch cases like NAME vs libNAME. Precedence goes to exact matches anyway, so this would be only in cases where that fails.